src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,176 @@
+					out.MaybeDie(err, "unable to execute the request: %v\n", err)
+					err = cl.MoveReplicas(cmd.Context(), ntp, replicas)
+					out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+				}
+			}
+			fmt.Print("Successfully submitted the request(s). Check the status with 'rpk cluster partitions move-status' or see the new assignments with 'rpk topic describe -p TOPIC'.\n")
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partition", "p", nil, "Partitions to move and new replica locations (repeatable)")
+	cmd.MarkFlagRequired("partition")
+	return cmd
+}
+
+// Expected input format is '0:1,2,3' or '1:1-0,2-0,3-0'
+// If '0:1,2,3' is passed, returns
+//
+// - '0'
+// - [{NodeID:1},{NodeID:2},{NodeID:3}]
+//
+// If '1:1-0,2-0,3-0' is passed, returns
+//
+// - '1'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parsePartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	pr := strings.Split(partition, ":")
+	if len(pr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if strings.Contains(pr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; unexpected \"/\"", partition)
+	}
+	replicasTmp := strings.Split(pr[1], ",")
+	for _, replicaCore := range replicasTmp {
+		nc := strings.Split(replicaCore, "-")
+		if len(nc) > 1 {
+			nodeID, err := strconv.Atoi(nc[0])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return pr[0], replicas, nil
+}
+
+// Expected input format is 'foo/0:1,2,3' or 'redpanda/controller/0:1-0,2-0,3-0'
+// If 'foo/0:1,2,3' is passed, returns
+//
+// - 'kafka/foo/0'
+// - [{NodeID:1,},{NodeID:2},{NodeID:3}]
+//
+// If 'redpanda/controller/0:1-0,2-0,3-0' is passed, returns
+//
+// - 'redpanda/controller/0'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parseTopicPartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	tpr := strings.Split(partition, ":")
+	if len(tpr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if !strings.Contains(tpr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; specify the topic name", tpr[0])
+	}
+	if strings.Count(tpr[0], "/") == 1 {
+		tpr[0] = "kafka/" + tpr[0]
+	}
+	replicasTmp := strings.Split(tpr[1], ",")
+	for _, replicaCore := range replicasTmp {
```

Comment:
You can avoid the replicasTmp and move strings.Split inline: `for _, replicaCore := range strings.Split(tpr[1], ",")`. Same comment for below.

Reply:
Fixed in https://github.com/redpanda-data/redpanda/commit/1595272ed458ac47621cdbcad4c550c99140d622